### DISCUSSION TITLE
CSV export support in WiFiExplorer Pro 3

### DISCUSSION THREAD

**Discussion by wifinigel** [joshschmelzle/lswifi#61] [Ideas]
It would be nice to inspect the CSV output in WiFiExplorer Pro 3. Have you thought about asking Adrian to add a decoder for lswifi output? It would be very cool to review the output in WFE Pro 3.

**Comment by joshschmelzle** [comment]
It could be interesting to come up with a common format and schema that can be shared amongst tools. I'm not sure that CSV is the right choice because for example I could output the raw bytes from the frame (which could be based on a beacon, probe response, or even a combination of both). Having the raw information would enable tools to use their own parser to determine the contents. For example, the parser for lswifi is not nearly as complete or robust as Adrian's.

**Reply by joshschmelzle** [reply]
Discussion is somewhat related to issue #21 brought up a while back. I'm not sure JSON is the right format either. lswifi was initially designed to just show a current snapshot of nearby networks. Analyzing changes over time is something I've thought about, but never sat down and prototyped anything. I think it would be interesting to improve the output lswifi for consumption by something else other than a human to enable certain ideas.

**Reply by wifinigel** [reply]
Josh, WFE Pro 3 already has a CSV import mechanism and supports multiple tools such as Airtool & Aruba Utilities. It only requires a snapshot of detected BSSIDs, not continuous data over time. Adrian would just need to know the column format of the existing lswifi CSV data export.

**Reply by joshschmelzle** [reply]
Oh! I did not know about that feature in WFE. This should just be a matter of lining up CSV format output then.

**Reply by joshschmelzle** [reply]
Hey @wifinigel, I talked with Adrian and the best solution here is for me to implement exporting to pcap. Once that's done, WFE, Wireshark, etc. can open lswifi's pcaps. So, I won't improve the CSV export for importing into WFE. Do you have any other ideas or use cases for CSV export?

**Reply by wifinigel** [reply]
No, I can't think of any other CSV use cases. I think the pcap approach is a great idea!
